### ISSUE TITLE
SC2016 heredoc + ${x:+expansion '${x}}

### ISSUE THREAD

**Issue by grooverdan** [koalaman/shellcheck#2383]
#### For bugs

- SC2016 
- online, 0.8.0 & 0.7.2
- [ X ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ X ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
password=bob
mysql "mysql" --binary-mode <<EOF
  create user 'user'@'%' ${password:+identified with 'mysql_native_password' by '${password}'};
EOF
```

#### Here's what shellcheck currently says:

```
In /tmp/p.sh line 4:
  create user 'user'@'%' ${password:+identified with 'mysql_native_password' by '${password}'};
                                                                                ^-----------^ SC2016: Expressions don't expand in single quotes, use double quotes for that.
```

#### Here's what I wanted or expected to see:

Nothing.

It seems there's already a SC2016 shell check excemption for heredocs, its the `${x:+'${x}'}` expansion that isn't covered in the same rules. Maybe its too specialized. Up to you. Thanks for the great program.
